utilities/checkpoint/checkpoint_impl.cc

```diff
@@ -59,7 +59,17 @@ Status CheckpointImpl::CreateCheckpoint(const std::string& checkpoint_dir,
       db_options.info_log,
       "Started the snapshot process -- creating snapshot in directory %s",
       checkpoint_dir.c_str());
-  std::string full_private_path = checkpoint_dir + ".tmp";
+
+  size_t final_nonslash_idx = checkpoint_dir.find_last_not_of('/');
+  if (final_nonslash_idx == std::string::npos) {
+    // npos means it's only slashes, which means it's the root directory, but it
+    // shouldn't be because we verified above the directory doesn't exist.
+    assert(false);
+    return Status::InvalidArgument("invalid checkpoint directory name");
+  }
+
+  std::string full_private_path =
```

Comment:
LGTM
How about print `full_private_path` to help with possible debugging?

Reply:
I can log it.